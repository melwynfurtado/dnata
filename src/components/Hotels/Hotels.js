import React from 'react'
import PropTypes from 'prop-types'
import Hotel from './Hotel'
import Pagination from '../Pagination'

const Hotels =({ hotelCount, hotels, handleOnPaginate, page, perPage }) => {
  let hotelJsx

  if (hotelCount === 0) {
    hotelJsx = [
      <p key="no-hotels" className="lead m-3">
        No hotels found. Please try to adjust your query or use one of the existing filters provided above.
      </p>
    ]
  } else {
    hotelJsx = [
      hotels.map(hotel => <Hotel key={hotel.EstablishmentId} hotel={hotel} />),
      <Pagination key="Pagination" { ...{ hotelCount, page, perPage, handleOnPaginate } } />
    ]
  }

  return <div className="hotels">{ hotelJsx }</div>
}

Hotels.propTypes = {
  hotelCount: PropTypes.number,
  hotels: PropTypes.arrayOf(
    PropTypes.shape({
      EstablishmentId: PropTypes.number.isRequired,
    })
  ),
  handleOnPaginate: PropTypes.func,
  page: PropTypes.number,
  perPage: PropTypes.number,
}

export default Hotels