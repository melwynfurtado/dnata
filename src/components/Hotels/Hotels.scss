.hotels {
  .sortbar {
    background-color: #e3f2fd;
  }  
}