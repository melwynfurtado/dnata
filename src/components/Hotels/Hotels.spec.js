import React from 'react'
import renderer from 'react-test-renderer'
import Hotels from './Hotels'

describe('Hotels Component', () => {
  const hotels = [{
    "Distance": 3.4872358588140977,
    "EstablishmentId": 2182944,
    "EstablishmentType": "Hotel",
    "Location": "Paris",
    "MinCost": 844,
    "Name": "Aberotel Montparnasse",
    "Stars": 3,
    "UserRating": 8,
    "UserRatingTitle": "Great",
    "UserRatingCount": 1,
    "ImageUrl": "https://i.t-rp.co.uk/ei/2/1/8/2/9/4/4/0_400_400.jpg",
    "ThumbnailUrl": "https://i.t-rp.co.uk/ei/2/1/8/2/9/4/4/0_80_80.jpg"
  }]

  it('renders correctly', () => {
    const tree = renderer
                  .create(<Hotels 
                      hotels={hotels} 
                      hotelCount={1} 
                      handleOnPaginate={()=>{}} 
                      handleOnSort={()=>{}} 
                      page={1} 
                      perPage={20} 
                      sortBy={{key: 'Stars', order: 'desc'}} />)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders "No hotels found. Please try to adjust your query or use one of the existing filters provided above."', () => {
    const tree = renderer
                  .create(<Hotels 
                    hotels={[]} 
                    hotelCount={0} 
                    handleOnPaginate={()=>{}} 
                    handleOnSort={()=>{}} 
                    page={1} 
                    perPage={20} 
                    sortBy={{key: 'Stars', order: 'desc'}} />)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })
})