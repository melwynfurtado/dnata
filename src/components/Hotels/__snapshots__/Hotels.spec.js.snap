// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotels Component renders "No hotels found. Please try to adjust your query or use one of the existing filters provided above." 1`] = `
<div
  className="hotels"
>
  <p
    className="lead m-3"
  >
    No hotels found. Please try to adjust your query or use one of the existing filters provided above.
  </p>
</div>
`;

exports[`Hotels Component renders correctly 1`] = `
<div
  className="hotels"
>
  <div
    className="hotel card mb-3"
  >
    <div
      className="card-body"
    >
      <div
        className="card-text"
      >
        <img
          alt="Aberotel Montparnasse"
          className="rounded float-md-left float-sm-left mr-3"
          src="https://i.t-rp.co.uk/ei/2/1/8/2/9/4/4/0_80_80.jpg"
        />
        <div
          className="float-md-left float-sm-left"
        >
          <h5
            className="card-title"
          >
            Aberotel Montparnasse
          </h5>
          <div
            className="stars"
          >
            <span
              className="fa fa-star checked"
            />
            <span
              className="fa fa-star checked"
            />
            <span
              className="fa fa-star checked"
            />
            <span
              className="fa fa-star false"
            />
            <span
              className="fa fa-star false"
            />
          </div>
          <div>
            Paris
              (
            <span
              className="fa fa-map-marker"
            />
             
            3.49
             miles away)
          </div>
        </div>
        <div
          className="float-md-right text-md-right float-sm-left text-sm-left"
        >
          <div>
            Great
             - 
            8
             / 10
          </div>
          <div
            className="mb-3 text-muted"
          >
            1
             reviews
          </div>
          <div
            className="font-weight-bold font-italic"
          >
            £
            844
             (per night)
          </div>
        </div>
      </div>
    </div>
  </div>
  <nav>
    <ul
      className="pagination float-right"
    >
      <li
        className="page-item prev disabled"
      >
        <a
          aria-label="Previous"
          className="page-link"
          href="#"
          onClick={[Function]}
          page={0}
        >
          <span
            aria-hidden="true"
            onClick={[Function]}
            page={0}
          >
            «
          </span>
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
      </li>
      <li
        className="page-item page-item-1 active"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
          page={1}
        >
          1
        </a>
      </li>
      <li
        className="page-item next disabled"
      >
        <a
          aria-label="Next"
          className="page-link"
          href="#"
          onClick={[Function]}
          page={2}
        >
          <span
            aria-hidden="true"
            onClick={[Function]}
            page={2}
          >
            »
          </span>
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </li>
    </ul>
    <div
      className="font-italic text-muted"
    >
      Page: 1 / 1
    </div>
  </nav>
</div>
`;
